### PR TITLE
User Registration/Cohort Member Creation

### PR DESCRIPTION
**Pull request by brvarner**
- Allows user to select between mentor and mentee registration via a stimulus file: `app/javascript/controllers/signup_controller`
- Created mentor and mentee forms, will consolidate into a single form later, both have internal forms that add the cohort attributes for...
- Creates a cohort_member instance when a user registers, the cohorts_attributes include the user's role, their capacity (nil if mentee role), and their cohort ID (for now)

### CODE REVIEW

File: app/javascript/controllers/signup_controller.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+// Connects to data-controller="signup"
+export default class extends Controller {
+  static targets = ["mentorForm", "menteeForm"];
+
+  connect() {
+    this.mentorFormTarget.hidden = true;
+  }
+
+  showForm({ params: { content } }) {
```
Comment (rezahusain) [inline app/javascript/controllers/signup_controller.js:11]: Love that you added the ability to switch between both forms depending on role, will make it a lot easier for routing down the road.

**Comment by rezahusain** [comment]
Whenever you do decide to consolidate the mentor and mentee forms, definitely use a template. It'll reduce a lot of redundancy!

File: app/views/users/registrations/_mentee_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                required: true %>
+        <%= f.input :profile_picture,
+                required: true %>
+        <%= f.input :skills_array,
+                required: true, multiple: true %>
+      </div>
+      <%= f.simple_fields_for :cohorts, CohortMember.new do |cm| %>
+        <%= cm.hidden_field :capacity, value: nil %>
+        <%= cm.hidden_field :role, value: 'Mentee' %>
+        <%= cm.hidden_field :cohort_id, value: 1 %>
```
Comment (rezahusain) [inline app/views/users/registrations/_mentee_form.html.erb:40]: Just wondering - will this continuously put every new Mentee into Cohort 1 no matter what?

File: app/views/users/registrations/_mentor_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                required: true %>
+        <%= f.input :profile_picture,
+                required: true %>
+        <%= f.input :skills_array,
+                required: true %>
+      </div>
+      <%= f.simple_fields_for :cohorts, CohortMember.new do |cm| %>
+        <%= cm.input :capacity, required: true %>
+        <%= cm.hidden_field :role, value: 'Mentor' %>
+        <%= cm.hidden_field :cohort_id, value: 1 %>
```
Comment (rezahusain) [inline app/views/users/registrations/_mentor_form.html.erb:40]: Same question here for a new Mentor.

**Review by rezahusain — APPROVED** [review]
Outside all of that it looks really good! Awesome job Brandon

File: app/javascript/controllers/signup_controller.js (hunk 1 above)
Comment (brvarner) [reply app/javascript/controllers/signup_controller.js:11]: Thank you bro!

**Review by rezahusain — APPROVED** [review]
Looks great Brandon! Approved

File: app/views/users/registrations/_mentee_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                required: true %>
+        <%= f.input :profile_picture,
+                required: true %>
+        <%= f.input :skills_array,
+                required: true, multiple: true %>
+      </div>
+      <%= f.simple_fields_for :cohorts, CohortMember.new do |cm| %>
+        <%= cm.hidden_field :capacity, value: nil %>
+        <%= cm.hidden_field :role, value: 'Mentee' %>
+        <%= cm.hidden_field :cohort_id, value: Cohort.first.id %>
```
Comment (MayaS1111) [inline app/views/users/registrations/_mentee_form.html.erb:40]: Thanks for making this dynamic!

File: app/views/users/registrations/_mentor_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                required: true %>
+        <%= f.input :profile_picture,
+                required: true %>
+        <%= f.input :skills_array,
+                required: true %>
+      </div>
+      <%= f.simple_fields_for :cohorts, CohortMember.new do |cm| %>
+        <%= cm.input :capacity, required: true %>
+        <%= cm.hidden_field :role, value: 'Mentor' %>
+        <%= cm.hidden_field :cohort_id, value: Cohort.first.id %>
```
Comment (MayaS1111) [inline app/views/users/registrations/_mentor_form.html.erb:40]: Same here *thumbs up*

**Review by MayaS1111 — APPROVED** [review]
Everything looks good to me. I know it took lots of time and effort, thanks!

**Pull request merged**